pybamm/expression_tree/binary_operators.py

```diff
@@ -115,20 +115,20 @@ def fold_add_subtract(array, types):
         ret = None
         if len(array) > 0:
             ret = array[0]
-            for child, t in zip(array[1:], types[1:]):
-                if t == pybamm.Addition:
+            for child, type in zip(array[1:], types[1:]):
```

Comment:
don't think we should use `type` as a variable name, as it is also a built-in function in python

Reply:
ah ok, will change to `typ` (was finding `t` confusing as it's also time)